server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
It appears that mongoengine just passes any kwargs you give it on to the pymongo connection:

http://docs.mongoengine.org/apireference.html#connecting

If that is true, we can employ a proper write_concern in one place that will be just as safe as what we had before. The reason I believe that passing your own write concern will override the replica set one I linked to above is that they call that setting the *default* write concern. The language means to me that it is the write concern used when no other write concern is specified.

There may be a way to test to find out for sure - if we were to make a replica set with 3 nodes and set the default write concern to 3, we could kill -9 one of them and see if the above call returns or not.

Another point I forgot to make earlier is that Pulp used to work well with replica sets. Issue 1 isn't going to be the first time someone ever tries this - it used to work and was tested by @jdob. We regressed later on when we started using MongoDB for Celery's result backend, which was unfortunate. It was also unfortunate that the issue was not reported for so long and that there was even a comment in the result backend code to that effect.